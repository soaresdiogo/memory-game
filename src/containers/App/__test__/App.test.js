import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import App from '../App';

import '@testing-library/jest-dom/extend-expect';

const AppComponent = <App />;

describe('App component', () => {
  test('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(AppComponent, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  test('matches snapshot', () => {
    const tree = renderer.create(AppComponent).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
