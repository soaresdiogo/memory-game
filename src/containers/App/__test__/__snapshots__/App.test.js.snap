// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches snapshot 1`] = `
<div
  className="MuiGrid-root Dashboard-root-4 MuiGrid-container MuiGrid-align-items-xs-center MuiGrid-justify-xs-center"
  data-testid="dashboard"
>
  <div
    className="MuiPaper-root DashboardMenu-root-5 MuiPaper-elevation1 MuiPaper-rounded"
    data-testid="dashboardMenu"
  >
    <h4
      className="MuiTypography-root DashboardMenu-title-6 MuiTypography-h4 MuiTypography-alignCenter"
      data-testid="dashboardMenuTitle"
    >
      Memory Game
    </h4>
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-outlinedSizeLarge MuiButton-sizeLarge MuiButton-fullWidth"
      data-testid="dashboardButton"
      href="/game"
      onBlur={[Function]}
      onClick={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        className="MuiButton-label"
      >
        Start Game
      </span>
    </a>
  </div>
</div>
`;
